test/advection_test/test_host.F90

```diff
@@ -569,6 +569,59 @@ subroutine test_host(retval, test_suites)
       end if
       !-------------------
 
+      !-------------------
+      !water-species tests:
+      !-------------------
+
+      !Check that being a water species defaults to False:
+      call const_props(index_liq)%is_water_species(check, errflg, errmsg)
+      if (errflg /= 0) then
+         write(6, '(a,i0,a,a,i0,/,a)') "ERROR: Error, ", errflg, " trying ",  &
+              "to get water_species prop for cld_liq index = ", index_liq,    &
+              trim(errmsg)
+         errflg_final = -1 !Notify test script that a failure occurred
+      end if
+      if (errflg == 0) then
+         if (check) then !Should be False
+            write(6, *) "ERROR: 'is_water_species' should default to False ", &
+                 "for all constituents unless set by host model."
+            errflg_final = -1 !Notify test script that a failure occured
+         end if
+      else
+         !Reset error flag to continue testing other properties:
+         errflg = 0
+      end if
+
+      !Check that setting a constituent to be a water species works
+      !as expected:
+      call const_props(index_liq)%set_water_species(.true., errflg, errmsg)
+      if (errflg /= 0) then
+         write(6, '(a,i0,a,a,i0,/,a)') "ERROR: Error, ", errflg, " trying ",  &
+              "to set water_species prop for cld_liq index = ", index_liq,    &
+              trim(errmsg)
+         errflg_final = -1 !Notify test script that a failure occurred
+      end if
+      if (errflg == 0) then
+         call const_props(index_liq)%is_water_species(check, errflg, errmsg)
+         if (errflg /= 0) then
+            write(6, '(a,i0,a,i0,/,a)') "ERROR: Error, ", errflg,             &
+                 " trying to get water_species prop for cld_liq index = ",    &
+                 index_liq, trim(errmsg)
+            errflg_final = -1 !Notify test script that a failure occurred
+         end if
+      end if
+      if (errflg == 0) then
+         if (.not.check) then !Should now be True
```

Comment:
```suggestion
         if (.not. check) then !Should now be True
```
Might need a space?

Reply:
Fortran actually doesn't care about that space, but I do personally agree that having the space improves readability, so I've gone ahead and added it.

Reply:
Ah, good to know!